flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/ResourceManagerJobMasterTest.java

```diff
@@ -145,6 +171,10 @@ public void testRegisterJobMasterWithFailureLeaderListener() throws Exception {
 		Future<RegistrationResponse> declineFuture = resourceManager.registerJobMaster(rmLeaderSessionId, jmLeaderSessionId, jobMasterAddress, unknownJobIDToHAServices);
 		RegistrationResponse response = declineFuture.get(5, TimeUnit.SECONDS);
 		assertTrue(response instanceof RegistrationResponse.Decline);
+
+		if (testingFatalErrorHandler.hasExceptionOccurred()) {
+			testingFatalErrorHandler.rethrowError();
+		}
```

Comment:
Yes I think this makes sense for all of the distributed components RM, JM and TM. It would  be good to instantiate default components which you can replace with a special version of a component in case you need it for a test.